### PR TITLE
feat: Add support up to 2 CD-ROMs on `r/virtual_machine`

### PR DESCRIPTION
**Pull request by ckhordiasma**
### Description

Increased the number of CD-ROMs that can be attached to `r/virtual_machine` to two (2). 

### Release Note

* `resource/virtual_machine`: Add support for two (2) CD-ROMs attached to a virtual machine. ([#1631](https://github.com/terraform-providers/terraform-provider-vsphere/pull/1631))

### References

Closes: #1164

### CODE REVIEW

**Comment by tenthirtyam** [comment]
Hi @ckhordiasma 👋,

Can you please:
- Sign the HashiCorp CLA
- Update the description following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming
- Mark the pull request as draft if not yet complete and ready for review/merge

You may wish to review the cdrom subresource, as there are notes in the `CdromRefreshOperation`.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 1]
```diff
@@ -4253,6 +4253,60 @@ resource "vsphere_virtual_machine" "vm" {
...
+
+  disk {
+    label = "disk0"
+    size  = 20
+  }
+  cdrom {
+    datastore_id  = "${data.vsphere_datastore.iso_datastore.id}"
+    path          = "${var.iso_path}"
+  }
+	cdrom {
```
Comment (tenthirtyam) [inline vsphere/resource_vsphere_virtual_machine_test.go:4298]: Minor indentation error.

**Comment by ckhordiasma** [comment]
```
  cdrom {
    datastore_id = data.vsphere_datastore.datastore.id
    path = "/${var.vm_name}/_deviceImage-0.iso"
  }
  cdrom {
    client_device = true
  }
```

with this PR, and with cdroms defined in the above order, I was able to get my cisco 8000v OVA to install and boot successfully. Still working on unit test case

**Comment by tenthirtyam** [comment]
Hi @ckhordiasma - is this still a work in progress?

I imagine that there is a needed update to the 'vsphere_virtual_machine` resource docs, too?

Ryan Johnson
Staff II Solutions Architect | VMware, Inc.

**Comment by ckhordiasma** [comment]
Hi Ryan,

Yes, sorry for delayed response. I was able to get it to work functionally
but was struggling to get a unit test working for it. I also need to update
the docs as you mentioned. Should be able to have something by next week.

**Comment by tenthirtyam** [comment]
Hi @ckhordiasma,

Thank for the update, no rush. I was just doing to due diligence during some spare time. Thanks for your contributions to the project.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

**Comment by ckhordiasma** [comment]
I was not able to get the unit tests running in my test environment, I would have had to make too many changes to the the dev environment that I am using at work. However I have the test case in there if there's like a pipeline that can test it out.

**Comment by tenthirtyam** [comment]
> I was not able to get the unit tests running in my test environment, I would have had to make too many changes to the the dev environment that I am using at work. However I have the test case in there if there's like a pipeline that can test it out.

Could you also include any related documentation updates to `r/vsphere_virtual_machine`?

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

**Comment by tenthirtyam** [comment]
Hi, @ckhordiasma - Setting the pull request to draft whilst updates to docs are made.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

**Comment by tenthirtyam** [comment]
Hi, @ckhordiasma - let me know if you're able to update the docs for this enhancement.

If not, that's okay. I can co-author those edits and work to add this to the next milestone (`v2.3.0`).

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Comment by ckhordiasma** [comment]
Hi ryan,

Sorry for the late reply! Forgot about the docs.. I will write something up
for it tomorrow

**Comment by tenthirtyam** [comment]
Thanks for the updates, @ckhordiasma - I'll review these.

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Review by tenthirtyam — APPROVED** [review]
Hi @ckhordiasma - 👋 

Thank you for this contribution and for including the `testAccResourceVSphereVirtualMachineConfigBasicCdromIsoAndClientCdrom` test . 

I've reviewed and tested multiple scenarios for this change with **_very excellent_** results.

Approved! 🚀 

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

**Pull request merged**